### PR TITLE
JS/Ruby/QL/Python: sync dbscheme fragments

### PR DESCRIPTION
**Pull request by aibaars**

### CODE REVIEW

File: javascript/ql/lib/semmlecode.javascript.dbscheme [hunk 1]
```diff
@@ -1,103 +1,75 @@
...
-
-folders(unique int id: @folder,
-        varchar(900) name: string ref);
-
-
-@container = @folder | @file ;
+/*- Files and folders -*/
 
+/**
+ * The location of an element that is not an expression or a statement.
```
Comment (asgerf) [inline javascript/ql/lib/semmlecode.javascript.dbscheme:6]: In JS this contains table the location of statements and expressions as well. Is that not the case in other languages?

**Review by asgerf — DISMISSED** [review]
JS 👍 provided that tests and evaluations all look good

**Review by RasmusWL — APPROVED** [review]
:+1: from the Python changes, which were mostly changing `varchar(900)` to `string` it seems.

I assume you've talked with the core team to check that `string` is the better option for the evaluator? (I don't know this myself, which is why I'm asking)

**Comment by aschackmull** [comment]
> 👍 from the Python changes, which were mostly changing `varchar(900)` to `string` it seems.
> 
> I assume you've talked with the core team to check that `string` is the better option for the evaluator? (I don't know this myself, which is why I'm asking)

In the dbscheme, `string` and `varchar` are synonymous (and the number attached is meaningless). The latter only exists for historical reasons, and you should really use `string` instead.

**Comment by aibaars** [comment]
> In the dbscheme, `string` and `varchar` are synonymous (and the number attached is meaningless). The latter only exists for historical reasons, and you should really use `string` instead.

As far as I know the "reprType" is not used at all, it could be anything. Perhaps we should just remove them all at some point.

**Pull request merged**